### PR TITLE
Adds 'resource type' prefix for related items in GL JS API docs

### PR DESCRIPTION
**Pull request by domlet**
Closes https://github.com/mapbox/mapbox-gl-js-docs/issues/243
Pairs with: https://github.com/mapbox/mapbox-gl-js-docs/pull/646

- Adds 'resource type' prefix for related items in GL JS API docs in GL JS API reference pages

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -92,7 +92,7 @@ const exported = {
      * @returns {string} The currently set access token.
      * @example
      * mapboxgl.accessToken = myAccessToken;
-     * @see [Display a map](https://www.mapbox.com/mapbox-gl-js/examples/)
+     * @see Example: Example: [Display a map](https://www.mapbox.com/mapbox-gl-js/example/simple-map/)
```
Comment (asheemmamoowala) [inline src/index.js:95]:
```suggestion
     * @see Example: [Display a map](https://www.mapbox.com/mapbox-gl-js/example/simple-map/)
```

File: src/index.js [hunk 2]
```diff
@@ -245,7 +245,7 @@ Debug.extend(exported, {isSafari, getPerformanceMetrics: PerformanceUtils.getPer
  * if (!mapboxgl.supported()) {
  *   alert('Your browser does not support Mapbox GL');
  * }
- * @see [Check for browser support](https://www.mapbox.com/mapbox-gl-js/example/check-for-support/)
+ * @see Example: Example: [Check for browser support](https://www.mapbox.com/mapbox-gl-js/example/check-for-support/)
```
Comment (asheemmamoowala) [inline src/index.js:248]:
```suggestion
 * @see Example: [Check for browser support](https://www.mapbox.com/mapbox-gl-js/example/check-for-support/)
```

File: src/source/worker_source.js [hunk 3]
```diff
@@ -74,7 +74,7 @@ export type WorkerDEMTileCallback = (err: ?Error, result: ?DEMData) => void;
  * implementation may also be targeted by the {@link Source} via
  * `dispatcher.getActor().send('source-type.methodname', params, callback)`.
  *
- * @see {@link Map#addSourceType}
+ * @see Example: {@link Map#addSourceType}
```
Comment (asheemmamoowala) [inline src/source/worker_source.js:77]: revert

File: src/style-spec/validate/latest.js [hunk 4]
```diff
@@ -5,7 +5,7 @@ import validateStyle from '../validate_style.min.js';
  * Validate a style against the latest specification. This method is optimized
  * to keep its bundle size small by refraining from requiring jslint or old
  * style spec versions.
- * @see validateStyleMin
+ * @see Example: validateStyleMin
```
Comment (asheemmamoowala) [inline src/style-spec/validate/latest.js:8]: revert

File: src/ui/camera.js [hunk 5]
```diff
@@ -168,7 +168,7 @@ class Camera extends Evented {
      * var center = map.getCenter();
      * // access longitude and latitude values directly
      * var {longitude, latitude} = map.getCenter();
-     * @see Tutorial: [Use Mapbox GL JS in a React app](https://docs.mapbox.com/help/tutorials/use-mapbox-gl-js-with-react/#store-the-new-coordinates)
+     * @see Tutorial: Tutorial: [Use Mapbox GL JS in a React app](https://docs.mapbox.com/help/tutorials/use-mapbox-gl-js-with-react/#store-the-new-coordinates)
```
Comment (asheemmamoowala) [inline src/ui/camera.js:171]:
```suggestion
     * @see Tutorial: [Use Mapbox GL JS in a React app](https://docs.mapbox.com/help/tutorials/use-mapbox-gl-js-with-react/#store-the-new-coordinates)
```

File: src/ui/map.js [hunk 6]
```diff
@@ -2303,8 +2303,8 @@ class Map extends Camera {
      *   }
      * });
      *
-     * @see [Create a hover effect](https://docs.mapbox.com/mapbox-gl-js/example/hover-styles/)
-     * @see Tutorial: [Create interactive hover effects with Mapbox GL JS](https://docs.mapbox.com/help/tutorials/create-interactive-hover-effects-with-mapbox-gl-js/)
+     * @see Example: [Create a hover effect](https://docs.mapbox.com/mapbox-gl-js/example/hover-styles/)
+     * @see Example: Tutorial: [Create interactive hover effects with Mapbox GL JS](https://docs.mapbox.com/help/tutorials/create-interactive-hover-effects-with-mapbox-gl-js/)
```
Comment (asheemmamoowala) [inline src/ui/map.js:2307]:
```suggestion
     * @see Tutorial: [Create interactive hover effects with Mapbox GL JS](https://docs.mapbox.com/help/tutorials/create-interactive-hover-effects-with-mapbox-gl-js/)
```

**Review by asheemmamoowala — CHANGES_REQUESTED** [review]

File: src/style-spec/validate/latest.js (hunk 4 above)
Comment (domlet) [reply src/style-spec/validate/latest.js]: Reverted in e0f554c65

File: src/source/worker_source.js (hunk 3 above)
Comment (domlet) [reply src/source/worker_source.js]: Reverted in e0f554c65

File: src/ui/camera.js [hunk 7]
```diff
@@ -801,7 +801,7 @@ class Camera extends Evented {
      * map.fitScreenCoordinates(p0, p1, map.getBearing(), {
      *   padding: {top: 10, bottom:25, left: 15, right: 5}
      * });
-     * @see Used by {@link BoxZoomHandler}
+     * @see API documentation for {@link BoxZoomHandler}
```
[the remark this answers is not in the document]
Comment (domlet) [reply src/ui/camera.js]: Reverted in e0f554c65

**Review by HeyStenson — APPROVED** [review]
🍊

**Review by asheemmamoowala — APPROVED** [review]

**Pull request merged**